src/components/settings/SettingsAccessTab/SettingsAccessLinkManagement.jsx

```diff
@@ -1,121 +1,145 @@
-          },
-          {
-            Header: 'Date created',
-            accessor: 'dateCreated',
-          },
-          {
-            Header: 'Usage',
-            accessor: 'usage',
-          },
-        ]}
-        additionalColumns={[
-          {
-            id: 'action',
-            Header: '',
-            /* eslint-disable react/prop-types */
-            Cell: ({ row }) => {
-              if (row.original.linkStatus === 'activated') {
-                return (
-                  <div className="d-flex flex-row-reverse">
-                    <Button onClick={() => handleCopyLink(row)} variant="link">Copy</Button>
-                    <Button onClick={() => handleDeactivateLink(row)} variant="link">Deactivate</Button>
-                  </div>
-                );
-              }
-              return null;
-            },
-            /* eslint-enable */
-          },
-        ]}
+    <>
+      <SettingsAccessTabSection
+        title="Access via Link"
+        checked={isLinkManagementEnabled}
+        onFormSwitchChange={handleLinkManagementFormSwitchChanged}
+        onCollapsibleToggle={handleLinkManagementCollapsibleToggled}
       >
-        <DataTable.TableControlBar />
-        <DataTable.Table />
-        <DataTable.EmptyTable content={loadingLinks ? 'Loading...' : 'No links found'} />
-      </DataTable>
-    </SettingsAccessTabSection>
-
+        <p>Generate a link to share with your learners.</p>
+        <DataTable
+          data={links}
+          itemCount={links.length}
+          // eslint-disable-next-line no-unused-vars
+          tableActions={(i) => (
+            // TODO: Consider making this its own component with logic
+            <StatefulButton
+              {...generateLinkButtonProps}
+              onClick={handleGenerateLinkSuccess}
+              disabledStates={!isLinkManagementEnabled ? Object.keys(generateLinkButtonProps.labels) : undefined}
+            />
+          )}
+          columns={[
+            {
+              Header: 'Link',
+              accessor: 'link',
+              Cell: LinkTableCell,
+            },
+            {
+              Header: 'Status',
+              accessor: 'isValid',
+              Cell: StatusTableCell,
+            },
+            {
+              Header: 'Date created',
+              accessor: 'created',
+              Cell: DateCreatedTableCell,
+            },
+            {
+              Header: 'Usage',
+              accessor: 'usageCount',
+              Cell: UsageTableCell,
+            },
+          ]}
+          additionalColumns={[
+            {
+              id: 'action',
+              Header: '',
+              Cell: props => <ActionsTableCell {...props} onDeactivateLink={handleDeactivatedLink} />,
+            },
+          ]}
+        >
+          <DataTable.TableControlBar />
+          <DataTable.Table />
+          <DataTable.EmptyTable content={loadingLinks ? 'Loading...' : 'No links found'} />
```

Comment:
interesting... looks like we want to use this pattern of loading when rendering emptyTable in other places too (I see we use one of these without the 'loading' smarts in explore catalog. Also maybe datatable can auto do 'loading..' by default in some way (just a thought)

Reply:
@binodpant There will soon be a `isLoading` prop that can be added to `DataTable` to show a loading spinner overlaying the table data (see [this PR](https://github.com/edx/paragon/pull/991) for its status). Once that it in place, I would imagine this component will use `isLoading` instead of changing up the content of `DataTable.EmptyTable`.

![image](https://user-images.githubusercontent.com/2828721/147269629-5e56e0bf-b542-4cb3-ace0-1f481beb465e.png)